src/main.cpp

```diff
@@ -2373,6 +2375,12 @@ void static UpdateTip(CBlockIndex *pindexNew) {
     nTimeBestReceived = GetTime();
     mempool.AddTransactionsUpdated(1);
     LogPrintf("UpdateTip: new best=%s  height=%d  log2_work=%.8g  tx=%lu  date=%d progress=%f nbits=%u algo=%d\n",chainActive.Tip()->GetBlockHash().ToString(), chainActive.Height(), log(chainActive.Tip()->nChainWork.getdouble())/log(2.0), (unsigned long)chainActive.Tip()->nChainTx, chainActive.Tip()->GetBlockTime(),Checkpoints::GuessVerificationProgress(chainActive.Tip()), chainActive.Tip()->nBits,GetAlgo(chainActive.Tip()->nVersion));
+    double nReward = ((double)GetBlockValue(chainActive.Tip(), 0))/100000000;
+    sumReward += nReward;
```

Comment:
If intention was to have whole money supply printed why not just use nMoneySupply from CBlockIndex.
Are you sure this will work? If I'm not mistaken this function is called only when there are new blocks, so the sum will be sum of the new blocks.

Reply:
This was used to produce a data file for plotting the coin emission as a function of time.